validator/sawtooth_validator/journal/publisher.py

```diff
@@ -227,24 +245,37 @@ def finalize_block(self, identity_signing_key, pending_batches):
                     LOGGER.debug("Abandoning block %s:" +
                                  "root state hash has invalid txn applied",
                                  builder)
+                    # Update the pending batch list to be all the
+                    # batches that passed validation to this point and
+                    # none of the ones that failed. It is possible that
+                    # this batch caused a future batch to fail so
+                    # we leave all of the batches that failed after this
+                    # one in the list.
+                    bad_batches.append(batch)
+                    pending_batches = [x for x in self._pending_batches
```

Comment:
This does not get put into the target array, it just overwrites the reference.  This use will lose pending batches.